basic/src/test/java/org/jboss/hal/testsuite/test/runtime/SimpleWebserviceEndpointTestCase.java

```diff
@@ -120,9 +120,13 @@ public void after() throws Exception {
     public static void afterClass() throws IOException, TimeoutException, InterruptedException {
         try {
             DEPLOYMENT_FILE.delete();
-            operations.writeAttribute(WEBSERVICES_ADDRESS, STATISTICS_ENABLED, originalStatisticsEnabledValue);
             administration.restartIfRequired();
             administration.reloadIfRequired();
+            if (originalStatisticsEnabledValue != null) {
+                operations.writeAttribute(WEBSERVICES_ADDRESS, STATISTICS_ENABLED, originalStatisticsEnabledValue);
+            } else {
+                throw new IllegalStateException("Unknown original value of webservices statistics-enabled attribute.");
+            }
```

Comment:
Why throwing exception? `null` just means that retrieving original value failed => `beforeClass` failed. Don't see any reason to throw exception as any thrown exception might be hiding the original one (after class should be called even when before class failed => you will suddenly see only exception from after class and not the one from before class) or is my assumption regarding before and after class behaviour.